masonry/src/widget/kurbo.rs

```diff
@@ -0,0 +1,406 @@
+        self.ctx.request_paint();
+    }
+
+    pub fn set_stroke_style(&mut self, stroke_style: Stroke) {
+        self.widget
+            .stroke
+            .get_or_insert_with(Default::default)
+            .style = stroke_style;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_stroke_brush(&mut self, stroke_brush: Brush) {
+        self.widget
+            .stroke
+            .get_or_insert_with(Default::default)
+            .brush = stroke_brush;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_stroke_brush_transform(&mut self, stroke_brush_transform: Option<Affine>) {
+        self.widget
+            .stroke
+            .get_or_insert_with(Default::default)
+            .brush_transform = stroke_brush_transform;
+        self.ctx.request_paint();
+    }
+}
+
+// MARK: IMPL WIDGET
+impl Widget for KurboShape {
+    fn on_pointer_event(&mut self, _ctx: &mut EventCtx, _event: &PointerEvent) {}
+    fn on_text_event(&mut self, _ctx: &mut EventCtx, _event: &TextEvent) {}
+    fn on_access_event(&mut self, _ctx: &mut EventCtx, _event: &AccessEvent) {}
+    fn on_status_change(&mut self, _ctx: &mut LifeCycleCtx, _event: &StatusChange) {}
+    fn lifecycle(&mut self, _ctx: &mut LifeCycleCtx, _event: &LifeCycle) {}
+
+    fn layout(&mut self, _ctx: &mut LayoutCtx, bc: &BoxConstraints) -> Size {
+        let size = self.shape.bounding_box().size();
+        if !bc.contains(size) {
+            warn!("The shape is oversized");
+        }
+        size
+    }
+
+    fn paint(&mut self, _ctx: &mut PaintCtx, scene: &mut Scene) {
+        let transform = self
+            .transform
+            .then_translate(-self.shape.bounding_box().origin().to_vec2());
+        if let Some(FillParams {
+            mode,
+            brush,
+            brush_transform,
+        }) = &self.fill
+        {
+            scene.fill(*mode, transform, brush, *brush_transform, &self.shape);
+        }
+        if let Some(StrokeParams {
+            style,
+            brush,
+            brush_transform,
+        }) = &self.stroke
+        {
+            scene.stroke(style, transform, brush, *brush_transform, &self.shape);
+        }
+    }
+
+    fn accessibility_role(&self) -> Role {
+        Role::GraphicsSymbol
+    }
+
+    fn accessibility(&mut self, _ctx: &mut AccessCtx) {}
+
+    fn children_ids(&self) -> SmallVec<[WidgetId; 16]> {
+        SmallVec::new()
+    }
+
+    fn make_trace_span(&self) -> Span {
+        trace_span!("KurboShape")
+    }
+}
+
+impl SvgElement for KurboShape {
+    fn origin(&self) -> Point {
+        self.shape.bounding_box().origin()
+    }
+
+    fn size(&self) -> Size {
+        self.shape.bounding_box().size()
+    }
+
+    fn set_origin(&mut self, _: Point) {
+        panic!("a shape does not support setting its origin after creation")
+    }
+
+    fn set_size(&mut self, _: Size) {
+        panic!("a shape does not support setting its size after creation")
+    }
```

Comment:
It bears mentioning again: the fact that these trait methods panic is a code smell. To be clear, I'm not saying they should be rewritten to not panic. Rather, this code is a hint that the broader design should be reconsidered.